# Preprocess of AV mutation mouse HammerSeq data
[Ming, X., Zhu, B. & Zhang, Z. Simultaneously measuring the methylation of parent and daughter strands of replicated DNA at the single-molecule level by Hammer-seq.](https://www.nature.com/articles/s41596-020-00488-9#citeas)

## Utilizing scripts from HammerSeq's paper to deduplicate seq data
``samtools view read1.fastq.sam | python deduplicate-HammerSeq-reads.py | gzip -c > deduplicated_reads.sam.gz``

## Extract methylation events from deduplicated_reads.sam.gz
``zcat deduplicated_reads.sam.gz | perl extract-events-from-sam-file.pl > events_info``

## Count frequencies by genomic coordinates
``perl count-events-from-sam-file.pl events_info``

## Calculate the maintenance ratio for the genomic regions of interests (ROIs)
``bedtools intersect -a RIO.bed -b events_count.bed -wo | perl count-events-by-ROI.pl > events_and_maintenance_by_ROI.txt``
***
# Calculate the genomic coverage
## Remove reads with mapping quality lower than 20 & only keep mapped reads
``samtools view -bF 4 -q 20 -@ 15 input.bam > output_clean.bam``
>-bF 4 只提取比对上的reads，-q 代表只留下比对质量大于20的reads

## Calculate the coverage in 1,000,000 windows size
``bedtools coverage -a mm10_1M_windows.bed -b sorted.bam > depth.txt``
![Genomic coverage of WT02]()
